app/screens/thread/thread.js

```diff
@@ -26,6 +26,18 @@ export default class Thread extends PureComponent {
 
     state = {};
 
+    shouldComponentUpdate(nextProps) {
+        const nextPostsIds = nextProps.posts.map((p) => p.id);
+        const prevPostsIds = this.props.posts.map((p) => p.id);
+        for (let i = 0; i < nextPostsIds.length; i++) {
```

Comment:
Maybe just iterate the posts instead of mapping and then iterating those arrays? It's probably much faster to do
``` javascript
if (nextProps.posts.length !== this.props.posts.length) {
    return true;
}

const length = nextProps.posts.length; // I'm still not convinced that calculating the length once is a significant speedup, but everywhere else seems to do it
for (let i = 0; i < length; i++) {
    if (nextProps.posts[i].id !== this.props.posts[i].id) {
        return true;
    }
}
```

Reply:
I can change it to this form, but the first validation on the length is actually wrong cause of the pending post

Reply:
That should cause a rerender though, shouldn't it? It's still a new post even if it's pending

Reply:
hehe I was looking at it from the areStatesEqual perspective doh